presto-main/src/main/java/io/prestosql/execution/SqlQueryExecution.java

```diff
@@ -571,6 +577,15 @@ public QueryInfo getQueryInfo()
         }
     }
 
+    public List<StageInfo> getAllStages()
+    {
+        SqlQueryScheduler scheduler = queryScheduler.get();
+        if (scheduler != null) {
+            return scheduler.getAllStages();
```

Comment:
stages can have children. I don't think it supports that, see `io.prestosql.execution.StageInfo#getAllStages`

Reply:
I see, changed to use `io.prestosql.execution.StageInfo#getAllStages` now, thanks